File: Cryptography/VigenereCipher.java
import java.util.Scanner;

public class VigenereCipher {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the plaintext: ");
        String plaintext = sc.nextLine();
        System.out.print("Enter the key: ");
        String key = sc.nextLine();
        
        String ciphertext = encrypt(plaintext, key);
        System.out.println("Ciphertext: " + ciphertext);
        
        String decryptedText = decrypt(ciphertext, key);
        System.out.println("Decrypted Text: " + decryptedText);
    }
    
    public static String encrypt(String plaintext, String key) {
        plaintext = plaintext.toUpperCase();
        key = key.toUpperCase();
        String ciphertext = "";
        int j = 0;
        for (int i = 0; i < plaintext.length(); i++) {
            char c = plaintext.charAt(i);
            if (c >= 'A' && c <= 'Z') {
                int shift = key.charAt(j) - 'A';
                char encrypted = (char) (((c - 'A' + shift) % 26) + 'A');
                ciphertext += encrypted;
                j = (j + 1) % key.length();
            } else {
                ciphertext += c;
            }
        }
        return ciphertext;
    }
    
    public static String decrypt(String ciphertext, String key) {
        ciphertext = ciphertext.toUpperCase();
        key = key.toUpperCase();
        String plaintext = "";
        int j = 0;
        for (int i = 0; i < ciphertext.length(); i++) {
            char c = ciphertext.charAt(i);
            if (c >= 'A' && c <= 'Z') {
                int shift = key.charAt(j) - 'A';
                char decrypted = (char) (((c - 'A' - shift + 26) % 26) + 'A');
                plaintext += decrypted;
                j = (j + 1) % key.length();
            } else {
                plaintext += c;
            }
        }
        return plaintext;
    }
}

File: Cryptography/vigenereCipher.cpp
#include <iostream>
#include <string>
#include <algorithm>

std::string encrypt(std::string plaintext, std::string key);
std::string decrypt(std::string ciphertext, std::string key);

int main() {
    std::string plaintext, key;
    std::cout << "Enter the plaintext: ";
    std::getline(std::cin, plaintext);
    std::cout << "Enter the key: ";
    std::getline(std::cin, key);
    
    std::string ciphertext = encrypt(plaintext, key);
    std::cout << "Ciphertext: " << ciphertext << std::endl;
    
    std::string decryptedText = decrypt(ciphertext, key);
    std::cout << "Decrypted Text: " << decryptedText << std::endl;
    
    return 0;
}

std::string encrypt(std::string plaintext, std::string key) {
    std::transform(plaintext.begin(), plaintext.end(), plaintext.begin(), ::toupper);
    std::transform(key.begin(), key.end(), key.begin(), ::toupper);
    std::string ciphertext = "";
    int j = 0;
    for (int i = 0; i < plaintext.length(); i++) {
        char c = plaintext[i];
        if (c >= 'A' && c <= 'Z') {
            int shift = key[j] - 'A';
            char encrypted = (((c - 'A' + shift) % 26) + 'A');
            ciphertext += encrypted;
            j = (j + 1) % key.length();
        } else {
            ciphertext += c;
        }
    }
    return ciphertext;
}

std::string decrypt(std::string ciphertext, std::string key) {
    std::transform(ciphertext.begin(), ciphertext.end(), ciphertext.begin(), ::toupper);
    std::transform(key.begin(), key.end(), key.begin(), ::toupper);
    std::string plaintext = "";
    int j = 0;
    for (int i = 0; i < ciphertext.length(); i++) {
        char c = ciphertext[i];
        if (c >= 'A' && c <= 'Z') {
            int shift = key[j] - 'A';
            char decrypted = (((c - 'A' - shift + 26) % 26) + 'A');
            plaintext += decrypted;
            j = (j + 1) % key.length();
        } else {
            plaintext += c;
        }
    }
    return plaintext;
}
